lib/dom_html.ml

```diff
@@ -166,13 +166,19 @@ end
 
 and keyboardEvent = object
   inherit event
-  method charCode : int optdef readonly_prop
-  method keyCode : int readonly_prop
-  method keyIdentifier : js_string t optdef readonly_prop
   method altKey : bool t readonly_prop
   method shiftKey : bool t readonly_prop
   method ctrlKey : bool t readonly_prop
   method metaKey : bool t readonly_prop
+  method location : int readonly_prop
+
+  method key : js_string t optdef readonly_prop
+  method code : js_string t optdef readonly_prop
+
+  method which : int optdef readonly_prop
```

Comment:
The multiple inheritance is needed to have a uniform internal representation of event handlers. The resulting "biggest event" is used in contravariant position in handlers, so we can use any of the events in its place.

I will see if we can separate them by providing multiple constructors in place of [`CE_client_closure`](https://github.com/ocsigen/eliom/blob/d24b99e13bb957de4e83a31636dffec72490dd83/src/lib/eliom_runtime.shared.mli#L60).